### PR TITLE
[BEAM-5663] Add Python 3.6 and Python 3.7 precommit test suites

### PR DESCRIPTION
**Pull request by RobbeSneyders**
This is is part of a series of PRs with goal to make Apache Beam PY3 compatible. The proposal with the outlined approach has been documented here: https://s.apache.org/beam-python-3.

This PR checks all unit tests for Python 3.6 and 3.7 to get an overview of remaining work.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by RobbeSneyders** [comment]
Run Python Precommit

**Comment by RobbeSneyders** [comment]
Run Python PreCommit

**Comment by RobbeSneyders** [comment]
R: @tvalentyn @markflyhigh

File: sdks/python/test-suites/tox/py37/build.gradle [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * Unit tests for Python 3.7
+ */
+
+plugins { id 'org.apache.beam.module' }
+applyPythonNature()
+
+// Required to setup a Python 3 virtualenv.
+project.ext.python3 = true
```
Comment (markflyhigh) [inline sdks/python/test-suites/tox/py37/build.gradle]: This is changed recently to have ability to configure a particular python version in virtualenv. Should be `pythonVersion = '3.7'`. See [the example](https://github.com/apache/beam/blob/master/sdks/python/test-suites/tox/py36/build.gradle#L27).

Comment (RobbeSneyders) [reply sdks/python/test-suites/tox/py37/build.gradle]: Thanks! I started on this before your PR was merged. It's fixed now.

File: sdks/python/test-suites/tox/py37/build.gradle [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * Unit tests for Python 3.7
+ */
+
+plugins { id 'org.apache.beam.module' }
+applyPythonNature()
+
+// Required to setup a Python 3 virtualenv.
+pythonVersion = '3.6'
```
Comment (markflyhigh) [inline sdks/python/test-suites/tox/py37/build.gradle]: should be `3.7` instead?

Comment (RobbeSneyders) [reply sdks/python/test-suites/tox/py37/build.gradle]: Sorry, too fast :) Updated.

**Comment by markflyhigh** [comment]
Run Python PreCommit

**Comment by markflyhigh** [comment]
Python precommit passed. Other two Jenkins failures are not related to this change. I can merge now.

**Comment by tvalentyn** [comment]
It looks like not all Jenkins can run Py37 suite yet. I think it is because only half of Jenkins workers use @yifanzou 's new VM image, the other half does not have Python37-dev dependencies installed which causes precommit flakes, see:
https://builds.apache.org/job/beam_PreCommit_Python_Commit/5704

`19:29:51     grpc_tools/_protoc_compiler.cpp:36:20: fatal error: Python.h: No such file or directory
19:29:51     compilation terminated.
19:29:51     error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
19:29:51     
19:29:51     ----------------------------------------
19:29:51 Command "/home/jenkins/jenkins-slave/workspace/beam_PreCommit_Python_Commit/src/build/gradleenv/1311307983/bin/python3.7 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-pnywbiyq/grpcio-tools/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-x0_of8g2/install-record.txt --single-version-externally-managed --compile --install-headers /home/jenkins/jenkins-slave/workspace/beam_PreCommit_Python_Commit/src/build/gradleenv/1311307983/include/site/python3.7/grpcio-tools" failed with error code 1 in /tmp/pip-install-pnywbiyq/grpcio-tools/
19:29:51 
19:29:51 > Task :beam-sdks-python-test-suites-tox-py37:setupVirtualenv FAILED`.

We may need to revert this and roll-forward without 3.7 support, or try to fix-forward and disable 3.7 tox suite for now.

**Comment by tvalentyn** [comment]
Sent https://github.com/apache/beam/pull/8292 to mitigate flakiness.

**Pull request merged**